functions.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Creates a connection to the collectorapp database
+ *
+ * @return PDO
+ */
+    function createDBConnection(): PDO {
+        $db = new PDO('mysql:host=db; dbname=collectorapp', 'root', 'password');
+        $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+        return $db;
+    }
+
+/**
+ * A query to the database that fetches the data in all the fields and creates a recipes assoc array
+ *
+ * @param $db
+ * @return array
+ */
+    function fetchAllRecipes($db): array {
+        $query = $db->prepare('SELECT `recipe`, `cuisine`, `time`, `link` FROM `recipes`;');
+        $query->execute();
+        $recipes = $query->fetchAll();
+        return $recipes;
+    }
+
+/**
+ * Creates recipe cards which have a name as well as stats on cuisine, time and a link to the recipe
+ *
+ * @param array $recipes
+ * @return string
+ */
+    function createRecipeCards(array $recipes): string {
```

Comment:
Caps on type hints i think

Reply:
Think I checked PS-12 and it doesn't say about caps